server/historian/packages/historian-base/src/runnerFactory.ts

```diff
@@ -47,7 +49,8 @@ export class HistorianResourcesFactory implements utils.IResourcesFactory<Histor
         const tenantCache = new historianServices.RedisTenantCache(redisClient);
         // Create services
         const riddlerEndpoint = config.get("riddler");
-        const riddler = new historianServices.RiddlerService(riddlerEndpoint, tenantCache);
+        const asyncLocalStorage = config.get("asyncLocalStorageInstance")[0];
```

Comment:
What happens when this is undefined? It looks like it's a non-optional argument in most places it's being passed through, but this can return undefined. For example, server/historian/packages/historian/src/www.ts isn't setting this config value.

Reply:
It could be possible the returned value is undefined here. We can change params in functions to optional to avoid service crushing due to manual mistakes.